dap-go.el

```diff
@@ -28,23 +28,46 @@
 ;;; Code:
 
 (require 'dap-mode)
+(require 'helm nil 'no-error)
 
-(defcustom dap-go-debug-program `("node" ,(expand-file-name "~/.extensions/go/out/src/debugAdapter/goDebug.js"))
+(defcustom dap-go-debug-program `("node"
+                                  ,(concat (file-name-directory (or load-file-name buffer-file-name))
+                                           "/bin/go/out/src/debugAdapter/goDebug.js"))
   "The path to the go debugger."
   :group 'dap-go
   :type '(repeat string))
 
-(defcustom dap-go-delve-path (expand-file-name "dlv" (expand-file-name "bin" (getenv "GOPATH")))
+(defcustom dap-go-delve-path (or (executable-find "dlv")
+                                 (expand-file-name "dlv" (expand-file-name "bin" (getenv "GOPATH"))))
   "The path to the delve command."
   :group 'dap-go
   :type 'string)
 
 (defun dap-go--populate-default-args (conf)
   "Populate CONF with the default arguments."
   (setq conf (pcase (plist-get conf :mode)
-	       ("auto" (dap-go--populate-auto-args conf))
-	       ("debug" (dap--put-if-absent conf :program (lsp-find-session-folder (lsp-session) (buffer-file-name))))
-	       ("exec" (dap--put-if-absent conf :program (read-file-name "Select executable to debug.")))))
+               ("auto" (dap-go--populate-auto-args conf))
+               ("debug" (dap--put-if-absent conf :program (lsp-find-session-folder (lsp-session) (buffer-file-name))))
+               ("exec" (dap--put-if-absent conf :program (read-file-name "Select executable to debug.")))
+               ("remote"
+                (prog2
+                    (dap--put-if-absent conf :program (lsp-find-session-folder (lsp-session) (buffer-file-name)))
+                    (dap--put-if-absent conf :port (string-to-number (read-string "Enter port: " "2345")))
+                  (start-process "dlv-attach" nil dap-go-delve-path
```

Comment:
Does it make sense to use `:program-to-start` instead of starting the process here? This will allow you to do `dap-debug-edit-template` and edit the template before running it. (I am fine with accepting as it is)

Reply:
updated! Thanks